File: 001/solution_001.py
#!/usr/bin/env python


def project_euler_001(ceiling):
    numbers   = xrange(1, ceiling)
    multiples = filter(lambda n: not n % 3 or not n % 5, numbers)
    solution  = sum(multiples)

    print solution


if __name__ == '__main__':
    project_euler_001(1000)

File: 001/solution_001.js
#! /usr/bin/env node

var project_euler_001 = function() {
    n = 1000
    numbers = Array.from({length: n - 1}, (_v, i) => i + 1)
    multiples = numbers.filter((x, _i, _a) => x % 3 == 0 || x % 5 == 0)
    solution = multiples.reduce((acc, x, _i, _a) => x + acc, 0)
    console.log("%d", solution)
}

project_euler_001()
